src/lib/dialog/dialog.ts

```diff
@@ -228,7 +228,9 @@ export class MdDialog {
     }
 
     if (componentOrTemplateRef instanceof TemplateRef) {
-      dialogContainer.attachTemplatePortal(new TemplatePortal(componentOrTemplateRef, null!));
+      dialogContainer.attachTemplatePortal(
+        new TemplatePortal<T>(componentOrTemplateRef, null!,
```

Comment:
Doesn't `null!` resolve to `never`? Seems like this needs a real `ViewContainerRef`